docker/development/debian/Dockerfile

```diff
@@ -150,29 +181,43 @@ RUN cd /tmp \
 
 # Dependencies
 
-## Boost [1.82.0]
+## Pybind11 [2.10.3]
```

Comment:
`2.10.4` is out: should we use it?

Also, let's `ARG` the version here too?

Reply:
It's not available in the ftp server yet so I think we can pass for now. Good call on the Arg.